network/server/tchannelthrift/node/service.go

```diff
@@ -240,34 +298,48 @@ func (s *service) FetchBlocksMetadata(tctx thrift.Context, req *rpc.FetchBlocksM
 		result.Elements[i] = blocksMetadata
 	}
 
+	s.metrics.fetchBlocksMetadata.Success.Inc(1)
+	s.metrics.fetchBlocksMetadata.Latency.Stop(sw)
 	return result, nil
 }
 
 func (s *service) Write(tctx thrift.Context, req *rpc.WriteRequest) error {
+	sw := s.metrics.write.Latency.Start()
 	ctx := tchannelthrift.Context(tctx)
 
 	if req.IdDatapoint == nil || req.IdDatapoint.Datapoint == nil {
+		s.metrics.write.Error.Inc(1)
+		s.metrics.write.Latency.Stop(sw)
```

Comment:
Maybe we should just log the latency if it's a successful operation?  I separated out success/error latency in `m3x/retry` package, however here maybe we just emit latency on success?

This could help reduce some of the duplicate code statements in this file.